rest/client/rest.go

```diff
@@ -711,6 +712,39 @@ func (c *communicatorImpl) DeleteCommitQueueItem(ctx context.Context, projectID,
 	return nil
 }
 
+func (c *communicatorImpl) EnqueueItem(ctx context.Context, id string) (int, error) {
+	info := requestInfo{
+		method:  put,
+		version: apiVersion2,
+		path:    fmt.Sprintf("/commitqueue/enqueue/%s", id),
+	}
+
+	resp, err := c.request(ctx, info, nil)
+	if err != nil {
+		return 0, err
+	}
+	defer resp.Body.Close()
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to read response")
+	}
+	if resp.StatusCode != http.StatusOK {
+		restErr := gimlet.ErrorResponse{}
+		if err = json.Unmarshal(bytes, &restErr); err != nil {
+			return 0, errors.Errorf("received an error but was unable to parse: %s", string(bytes))
+		}
+
+		return 0, restErr
+	}
+
+	position, err := strconv.Atoi(string(bytes))
```

Comment:
shouldn't the result be in json rather than a single integer?

Reply:
done.